### PR TITLE
WIP Floating player on scroll

### PR DESCRIPTION
**Pull request by kauffj**
see #4932 for context, @Ruk33 is looking for feedback on this

### CODE REVIEW

**Review by DispatchCommit — CHANGES_REQUESTED** [review]
Listening to the scroll event can lead to performance issues.

At the minimum, the event listener should be a passive event listener:
https://developer.mozilla.org/en-US/docs/Web/API/EventTarget/addEventListener

But the *best* solution would be to use an intersection observer instead:
https://developer.mozilla.org/en-US/docs/Web/API/Intersection_Observer_API

Intersection observers is what bitwave uses to popout it's player on scroll:

https://github.com/bitwave-tv/bitwave/blob/40dc640fde12cf26aa320006ec259d4e66aeb6dc/pages/_watch/index.vue#L21-L30

https://github.com/bitwave-tv/bitwave/blob/40dc640fde12cf26aa320006ec259d4e66aeb6dc/pages/_watch/index.vue#L376-L378

File: ui/page/file/view.jsx [hunk 1]
```diff
@@ -44,10 +46,32 @@ function FilePage(props: Props) {
...
+      doSetPlayingUri(uri, 'markdown');
+    }
+  }, [uri]);
+
+  React.useEffect(() => {
+    if (!floatingPlayerEnabled) {
+      return;
+    }
+
+    window.addEventListener('scroll', onScroll);
```
Comment (DispatchCommit) [inline ui/page/file/view.jsx:68]: https://developers.google.com/web/updates/2016/06/passive-event-listeners

**Comment by kauffj** [comment]
@Ruk33 do you want to come back to this?

**Comment by Ruk33** [comment]
Hello @kauffj sure, I was finishing the release date input now I'll continue with this one. Thanks for following up!

**Comment by seanyesmunt** [comment]
@Ruk33 I'm going to close this one while I am cleaning up the repo a bit. This would be a great feature to have though! Feel free to re-open this PR when you start working on it again.

**Pull request closed without merging**